cvmfs/util.cc

```diff
@@ -886,6 +886,35 @@ bool HasPrefix(const string &str, const string &prefix,
 }
 
 
+bool HasSuffix(
```

Comment:
What about:
``` C++
#include <algorithm>

struct IgnoreCaseComperator {
  bool operator()(const std::string::value_type a,
                  const std::string::value_type b) const {
    return std::tolower(a) == std::tolower(b);
  }
};

bool HasSuffix(const std::string &value,
               const std::string &ending,
               const bool         ignore_case = false) {
  if (ending.size() > value.size()) return false;
  const IgnoreCaseComperator icmp;
  return (ignore_case)
           ? std::equal(ending.rbegin(), ending.rend(), value.rbegin(), icmp)
           : std::equal(ending.rbegin(), ending.rend(), value.rbegin());
}
```

Less lines, easier to understand, use of STL and probably faster, because it only checks `ignore_case` once and potentially inlines the rest.